src/main/java/net/mcreator/minecraft/ElementUtil.java

```diff
@@ -45,6 +47,33 @@ public static Predicate<DataListEntry> typeMatches(String type) {
 		return e -> type.equals(e.getType());
 	}
 
+	/**
+	 * Loads a list of entries, with optional custom entries from the given mod element types,
+	 * and with an optional filter for the entry type.
+	 *
+	 * <p>NOTE: custom entries cannot specify a type yet, so the type filter will remove any custom entry</p>
+	 *
+	 * @param workspace The current workspace
+	 * @param dataList The datalist from which to load the entries
+	 * @param typeFilter If present, only entries whose type matches this parameter are loaded
+	 * @param customEntryProviders The string id of the mod element types that provide custom entries
+	 *
+	 * @return All entries from the given data list and the given mod element types, matching the optional filter
+	 */
+	public static List<DataListEntry> loadDataListAndElements(Workspace workspace, String dataList,
+			@Nullable String typeFilter, @Nullable String... customEntryProviders) {
+		List<DataListEntry> retval = DataListLoader.loadDataList(dataList);
+
+		if (customEntryProviders != null)
+			retval.addAll(getCustomElements(workspace,
+					me -> Arrays.asList(customEntryProviders).contains(me.getTypeString())));
+
+		if (typeFilter != null)
+			return retval.stream().filter(typeMatches(typeFilter)).toList();
+
+		return retval.stream().filter(e -> e.isSupportedInWorkspace(workspace)).sorted().toList();
```

Comment:
> We usually show all elements, even if the generator does not support them, at least we did so far.

This is consistent with how the `openEntrySelector` does it

> I am also wondering if maybe all loadAllXYZ method bodies should be replaced with calls to loadDataListAndElements so methods remain for convenience, but do not re-implement the same logic maybe?

Good point

Reply:
Ok, then go with this option, but then indeed all loadAllXYZ should call this method so we are consistent with what entries are shown and what not in relation to the generator coverage of entries.

Reply:
An issue I found with replacing some of these methods is how the entries are sorted. For example if I replace the `loadAllTabs` method, the tabs are sorted alphabetically, which doesn't happen now.

Would it be a good idea to add a "sorted" parameter to the method, so that other dropdowns aren't affected?

Reply:
I think that them being sorted now is ok, as long as custom entries are on the top

Reply:
I ultimately decided to add the "sorted" parameter anyway, as the alphabetic sort is unwieldy for some selectors (especially potions for brewing recipes). This also lets us skip the sorting during tests, where it doesn't matter anyway